src/dev/server.ts

```diff
@@ -192,16 +193,18 @@ export function createDevServer(nitro: Nitro): NitroDevServer {
 
   // Main worker proxy
   const proxy = createProxy();
-  proxy.proxy.on("proxyReq", (proxyReq, req) => {
+  proxy.proxy.on("proxyReq", (proxyReq, req: IncomingMessage) => {
     const proxyRequestHeaders = proxyReq.getHeaders();
     if (req.socket.remoteAddress && !proxyRequestHeaders["x-forwarded-for"]) {
       proxyReq.setHeader("X-Forwarded-For", req.socket.remoteAddress);
     }
     if (req.socket.remotePort && !proxyRequestHeaders["x-forwarded-port"]) {
       proxyReq.setHeader("X-Forwarded-Port", req.socket.remotePort);
     }
-    if (req.socket.remoteFamily && !proxyRequestHeaders["x-forwarded-proto"]) {
-      proxyReq.setHeader("X-Forwarded-Proto", req.socket.remoteFamily);
+    if (!proxyRequestHeaders["x-forwarded-proto"]) {
+      const url = new URL(req.url);
```

Comment:
`req.url` is a relative URL like `/foo`, we cannot parse it this way to infer proto